offload/plugins-nextgen/cuda/src/rtl.cpp

```diff
@@ -916,6 +916,11 @@ struct CUDADeviceTy : public GenericDeviceTy {
     return Plugin::check(Res, "error in cuStreamWaitEvent: %s");
   }
 
+  // TODO: This should be implementable on CUDA
+  Expected<bool> hasPendingWorkImpl(AsyncInfoWrapperTy &AsyncInfo) override {
+    return true;
```

Comment:
`cuStreamQuery` should do it, I can implement and test it in a follow up PR if you'd like

Reply:
That would be great. I didn't want to blindly code cuda library stuff.